src/shims/native_lib/trace/stub.rs

```diff
@@ -12,13 +14,17 @@ impl Supervisor {
         _: T,
         f: impl FnOnce() -> InterpResult<'tcx, crate::ImmTy<'tcx>>,
     ) -> InterpResult<'tcx, (crate::ImmTy<'tcx>, Option<super::MemEvents>)> {
+        let _g = SUPERVISOR.lock().unwrap();
         f().map(|v| (v, None))
     }
 }
 
 #[allow(dead_code, clippy::missing_safety_doc)]
-#[inline(always)]
 pub unsafe fn init_sv() -> Result<(), !> {
+    let mut sv_guard = SUPERVISOR.lock().unwrap();
+    if sv_guard.is_none() {
+        *sv_guard = Some(Supervisor);
+    }
     Ok(())
```

Comment:
Shouldn't this return `Err`, to trigger the error that we are using fallback mode?

Reply:
Ah, fair point